interface/ceed-basis.c

```diff
@@ -1599,22 +1599,25 @@ int CeedBasisApplyAtPoints(CeedBasis basis, CeedInt num_points, CeedTransposeMod
       CeedCall(CeedVectorGetArrayRead(basis->vec_chebyshev, CEED_MEM_HOST, &chebyshev_coeffs));
       CeedCall(CeedVectorGetArrayRead(x_ref, CEED_MEM_HOST, &x_array_read));
       CeedCall(CeedVectorGetArrayWrite(v, CEED_MEM_HOST, &v_array));
+
       switch (eval_mode) {
         case CEED_EVAL_INTERP: {
-          CeedScalar tmp[2][num_comp * CeedIntPow(Q_1d, dim)], chebyshev_x[Q_1d];
+          CeedInt    pre = num_comp * CeedIntPow(Q_1d, dim - 1), post = 1;
+          CeedScalar tmp[2][num_comp * num_points * CeedIntPow(Q_1d, dim)];
 
           // ---- Values at point
-          for (CeedInt p = 0; p < num_points; p++) {
-            CeedInt pre = num_comp * CeedIntPow(Q_1d, dim - 1), post = 1;
-
-            for (CeedInt d = 0; d < dim; d++) {
-              // ------ Tensor contract with current Chebyshev polynomial values
-              CeedCall(CeedChebyshevPolynomialsAtPoint(x_array_read[p * dim + d], Q_1d, chebyshev_x));
-              CeedCall(CeedTensorContractApply(basis->contract, pre, Q_1d, post, 1, chebyshev_x, t_mode, false,
-                                               d == 0 ? chebyshev_coeffs : tmp[d % 2], d == (dim - 1) ? &v_array[p * num_comp] : tmp[(d + 1) % 2]));
-              pre /= Q_1d;
-              post *= 1;
+          for (CeedInt d = 0; d < dim; d++) {
+            CeedScalar interp_1d[num_points * Q_1d];
+
+            // ------ Build interp matrix
+            for (CeedInt p = 0; p < num_points; p++) {
+              CeedCall(CeedChebyshevPolynomialsAtPoint(x_array_read[p * dim + d], Q_1d, &interp_1d[p * Q_1d]));
             }
+            // ------ Tensor contract with current Chebyshev polynomial values
+            CeedCall(CeedTensorContractApply(basis->contract, pre, Q_1d, post, num_points, interp_1d, t_mode, false,
+                                             d == 0 ? chebyshev_coeffs : tmp[d % 2], d == (dim - 1) ? v_array : tmp[(d + 1) % 2]));
+            pre /= Q_1d;
+            post *= 1;
```

Comment:
I think something has to be screwy here.

The 1d contraction works. 2D is wrong.

 I'm trying to contract from Q_1d * Q_1d (for example) down to 1 * num_points

Reply:
`CeedTensorContractStridedApply` might be what I need here?

Reply:
No that won't work either.

I need to go Q_1d * Q_1d -> Q_1d * num_points -> 1 * num_points

or in 3D
Q_1d * Q_1d * Q_1d-> Q_1d * Q_1d * num_points -> Q_1d * num_points -> 1 * num_points

Our current tensor contractions don't do that.

Reply:
Maybe setting post = Q_1d^dim-1 and decreasing it?